Add clock funtionality

matiasperz opened the pull request:
Add clock functionality so that ticker could be paused and restarted.

clementroche commented:
hello @matiasperz, thank you for your PR, as first sight this look interesting! I need to look at the code more in depth. I swear to do that over the weekend. To better understand, your idea it to restart Tempus to 0 once the page is visually ready?

matiasperz commented:
@clementroche that's right! My first thought was to initiate Tempus paused and play it when page is ready, but:
 - I get that Tempus is meant to be instanced just once, and it does it internally.
 - Since it's being instanced internally can't pass a config to it like `{ autoStart: false }`.
 - Starting it paused as default adds complexity to it's usage and might lead to confusion.
 
So that's what takes me to the idea of being able to reset it's elapsed time just before adding the gsap ticker to the Tempus `raf`.

clementroche commented:
Yo @matiasperz, I reviewed the PR and pushed some code. I've moved Clock into its own file, added documentation and simplified types.

Let me know if it's ready to be merged on your side.

matiasperz commented:
I've just pushed a fix for this error where I get negative delta time (equivalent to `-elapsedTime`) for the first tick after hitting `pause` and then `play` without a reset. Since the relative `Framerate` calculates their relative time based on adding the master clock delta to their internal elapsed time, it was delaying the UI update after play, until the running clock compensates for this negative value.

https://github.com/user-attachments/assets/c343aea6-43cd-4228-b59e-753401571251

The last commit ID with this error was `d1e5d81`.

To fix the issue, I've introduced some new time variables comparable to the [ThreeJS timer class](https://github.com/mrdoob/three.js/blob/master/examples/jsm/misc/Timer.js), which has proven to work, and simplified the `reset` mechanism.

Another key change from the last commit is the removal of `performance.now()` to set `clock._startTime`. The `requestAnimationFrame(elapsed)` value comes from [document.timeline.currentTime](https://developer.mozilla.org/en-US/docs/Web/API/AnimationTimeline/currentTime) and is not in sync with the `performance.now()`, so to prevent negative delta numbers, I just rely on the first one.

The clock API is still the same tho, ready to merge on my side 🫡.

clementroche commented on packages/core/src/tempus.ts:
```diff
@@ -181,6 +181,8 @@ class TempusImpl {
     const elapsed = this.clock.time
     const deltaTime = this.clock.deltaTime
 
+    console.log('raf >>>', { elapsed, deltaTime })
```
Can you remove the console log ?

matiasperz replied:
-1 aura point, fixed 😅

clementroche commented:
It's merged, thank you @matiasperz
